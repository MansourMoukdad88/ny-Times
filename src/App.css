.content {
  display: flex;
  align-items: center;
  background-color: #fafafa;
  padding: 20px 15px;
  margin: 0 10%;
}
.circle {
  width: 50px;
  min-height: 50px;
  border-radius: 100%;
  background-color: gray;
}
.date-content-container {
  display: flex;
  width: 70%;
  flex-direction: column;
}
.date-author {
  display: flex;
  flex-direction: row;
  margin-left: 10px;
  justify-content: space-between
  
}
.date-author span {
  color: rgb(172, 170, 170);
  display:flex;
  font-size: 12px;
}
.author {
  margin-right: 50px;
  text-overflow: ellipsis;
  width: 160px;
  font-size: 12px
}
.articleTitle {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}
.articleTitle {
  overflow: hidden;
  margin-left: 10px;
}
i {
  font-size:48px;
  margin-right: 7px;
  padding-top: 4px;
}
.fa-ellipsis-v {
  display: none;
}
.fa-bars {
  display: none
}
a {
  text-decoration: none !important;
  color: black;
}
.next {
  margin-left: 70px;
}


/* Navbar */
.nav-container {
  background-color: #47e4c2;
  width: 100%;
  min-height: 70px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0px 20px;
}
.nav-container {
  font-size: 25px;
  color:white;
}



/* Media Queries */
@media only screen and (max-width: 465px) {
  .fa-ellipsis-v {
    display: block;
  }
}
@media only screen and (max-width: 465px) {
  .fa-bars {
    display: block;
  }
}
@media only screen and (max-width: 575px) {
  .content {
    display: flex;
    align-items: center;
    background-color: #fafafa;
    padding: 20px 5px;
    margin: 0;
    }
}
@media only screen and (max-width: 465px) {
  .circle {
    width: 50px;
    min-height: 50px;
    border-radius: 50%;
    background-color: gray;
  }
}
@media only screen and (max-width: 400px) {
  .author {
    margin-right: 50px;
    text-overflow: ellipsis;
    width: 150px;
    font-size: 12px;
  }
}