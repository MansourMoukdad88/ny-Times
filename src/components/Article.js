import React, { Component} from 'react';
import ArticleList from './ArticleList';
import axios from "axios"

class Article extends Component {
  constructor(){
    super()
    this.state={
      data:[]
    }
  }
  componentDidMount=()=>{
    axios.get("https://api.nytimes.com/svc/mostpopular/v2/mostviewed/all-sections/7.json?api-key=qG808Xn6G5zAb6Wm0e6hS52JQjS6L2Er")
    .then(response => {
      console.log(response.data.results)
      this.setState({
        data:response.data.results
    })
    })
  }
  render(){
    //console.log("data",this.state.data)
    return (
      <div>
        {this.state.data.map((ele, i)=>{
          return <ArticleList key={i} ele={ele}/>
        })}              
      </div>
    )
  }
}

export default Article;