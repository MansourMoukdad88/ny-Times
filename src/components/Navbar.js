import React from 'react';

const Navbar = () => {
  return (
    <div>
      <div className="nav-container">
        <i className="fa fa-bars"></i>
        <p>NY Times Most Popular</p>        
        <i className="fa fa-search"></i>
        <i className="fa fa-ellipsis-v"></i>
      </div>
    </div>
  )
}

export default Navbar;