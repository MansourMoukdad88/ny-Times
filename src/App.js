import React from 'react';
import Navbar from "./components/Navbar";
import Article from './components/Article';
import "./App.css"


const App = () => {
  return (
    <div className="App">
      <Navbar/>
      <Article/>
    </div>
  );
}

export default App;
