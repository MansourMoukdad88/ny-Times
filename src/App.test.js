import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Navbar from './components/Navbar';
import Article from './components/Article';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';


describe('<App />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should matche the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render one <Navbar /> components', () => {
    expect(wrapper.find(Navbar)).toHaveLength(1);
  });
  it('should render one <Article /> components', () => {
    expect(wrapper.find(Article)).toHaveLength(1);
  });
});