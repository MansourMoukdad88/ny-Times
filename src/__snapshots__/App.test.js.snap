// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should matche the snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="nav-container"
    >
      <i
        className="fa fa-bars"
      />
      <p>
        NY Times Most Popular
      </p>
      <i
        className="fa fa-search"
      />
      <i
        className="fa fa-ellipsis-v"
      />
    </div>
  </div>
  <div />
</div>
`;
